File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long long, vector<long long>, greater<long long>> pq;
        for(auto n : nums) pq.push(n);
        int op = 0;
        while(pq.size() && pq.top() < k) {
            long x = pq.top(); pq.pop();
            long y = pq.top(); pq.pop();

            pq.push(2*x + y);
            op++;
        }
        return op;
    }
};

File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        for(int i : nums)
            pq.add((long)i);

        int op = 0;
        while(pq.size() > 1) {
            long x = pq.remove();
            if(x >= k) return op;
            long y = pq.remove();

            pq.add(Math.min(x,y) * 2L + Math.max(x,y));
            op++;
        }
        return op;
    }
}